### PR TITLE
Adds python 3.8.9, 3.9.3 and 3.9.4, closes #3439

### LINKED ISSUE #3439: python 3.8.9 is not found
**Package ecosystem**

- All python: `pip`, `poetry`, etc

**🕹 Bonus points: Smallest manifest that reproduces the issue**

```toml
# pyproject.toml

[build-system]
requires = ["poetry-core>=1.0.0"]
build-backend = "poetry.core.masonry.api"

[tool.poetry]
name = "a"
description = "b"
version = "0.1.0"
authors = []

license = "MIT"


[tool.poetry.dependencies]
python = "3.8.9"

[tool.poetry.dev-dependencies]

```

It is failing:

```
Dependabot detected the following Python requirements for your project: '[[#<Dependabot::Python::Requirement:0x000055568ad26c08 @requirements=[["=", #<Dependabot::Python::Version 3.8.9>]]>]]'.

Currently, the following Python versions are supported in Dependabot: 3.9.2, 3.9.1, 3.9.0, 3.8.8, 3.8.7, 3.8.6, 3.8.5, 3.8.4, 3.8.3, 3.8.2, 3.8.1, 3.8.0, 3.7.10, 3.7.9, 3.7.8, 3.7.7, 3.7.6, 3.7.5, 3.7.4, 3.7.3, 3.7.2, 3.7.1, 3.7.0, 3.6.13, 3.6.12, 3.6.11, 3.6.10, 3.6.9, 3.6.8, 3.6.7, 3.6.6, 3.6.5, 3.6.4, 3.6.3, 3.6.2, 3.6.1, 3.6.0, 3.5.10, 3.5.8, 3.5.7, 3.5.6, 3.5.5, 3.5.4, 3.5.3, 2.7.18, 2.7.17, 2.7.16, 2.7.15, 2.7.14, 2.7.13.

```
Link: https://github.com/wemake-services/wemake-django-template/issues/1529

But, `python3.8.9` is released: https://www.python.org/downloads/release/python-389/

### PR DESCRIPTION
**Pull request by sobolevn**
I am not sure if I also have to tweak some infra to test it 🤔

### CODE REVIEW

**Comment by sobolevn** [comment]
Looks like `pyenv` is not ready to support `3.8.9` yet:

```
        Dependabot::SharedHelpers::HelperSubprocessFailed:
         python-build: definition not found: 3.9.3

         See all available versions with `pyenv install --list'.
```

**Comment by sobolevn** [comment]
Now it looks like it should do it 👍

File: Dockerfile [hunk 1]
```diff
@@ -69,10 +69,10 @@ RUN apt-add-repository ppa:brightbox/ruby-ng \
 ENV PYENV_ROOT=/usr/local/.pyenv \
   PATH="/usr/local/.pyenv/bin:$PATH"
 RUN git clone https://github.com/pyenv/pyenv.git /usr/local/.pyenv \
-  && cd /usr/local/.pyenv && git checkout 9ee109b66148bc39a685926050b7b56cb4bb184b && cd - \
-  && pyenv install 3.9.2 \
+  && cd /usr/local/.pyenv && git checkout 32ba5d5e617b7d7fb1c9b79c1c0e8f1cfe642cd1 && cd - \
```
Comment (Parnassius) [inline Dockerfile:72]: Version 1.2.25 was just released
```suggestion
  && cd /usr/local/.pyenv && git checkout v1.2.25 && cd - \
```

Comment (sobolevn) [reply Dockerfile]: Thanks!

File: Dockerfile [hunk 2]
```diff
@@ -69,10 +69,10 @@ RUN apt-add-repository ppa:brightbox/ruby-ng \
 ENV PYENV_ROOT=/usr/local/.pyenv \
   PATH="/usr/local/.pyenv/bin:$PATH"
 RUN git clone https://github.com/pyenv/pyenv.git /usr/local/.pyenv \
-  && cd /usr/local/.pyenv && git checkout 9ee109b66148bc39a685926050b7b56cb4bb184b && cd - \
-  && pyenv install 3.9.2 \
+  && cd /usr/local/.pyenv && git checkout v1.2.25 && cd - \
+  && pyenv install 3.9.3 \
```
Comment (Parnassius) [inline Dockerfile:73]:
```suggestion
  && cd /usr/local/.pyenv && git checkout v1.2.26 && cd - \
  && pyenv install 3.9.4 \
```

File: Dockerfile [hunk 3]
```diff
@@ -69,10 +69,10 @@ RUN apt-add-repository ppa:brightbox/ruby-ng \
...
   && pyenv install 2.7.18 \
-  && pyenv global 3.9.2
+  && pyenv global 3.9.3
```
Comment (Parnassius) [inline Dockerfile:75]:
```suggestion
  && pyenv global 3.9.4
```

File: python/helpers/build [hunk 4]
```diff
@@ -17,4 +17,4 @@ cp -r \
 
 cd "$install_dir"
 PYENV_VERSION=2.7.18 pyenv exec pip install -r "requirements.txt"
-PYENV_VERSION=3.9.2  pyenv exec pip install -r "requirements.txt"
+PYENV_VERSION=3.9.3  pyenv exec pip install -r "requirements.txt"
```
Comment (Parnassius) [inline python/helpers/build:20]:
```suggestion
PYENV_VERSION=3.9.4  pyenv exec pip install -r "requirements.txt"
```

File: python/lib/dependabot/python/python_versions.rb [hunk 5]
```diff
@@ -4,14 +4,14 @@ module Dependabot
   module Python
     module PythonVersions
       PRE_INSTALLED_PYTHON_VERSIONS = %w(
-        3.9.2 2.7.18
+        3.9.3 2.7.18
```
Comment (Parnassius) [inline python/lib/dependabot/python/python_versions.rb:7]:
```suggestion
        3.9.4 2.7.18
```

File: python/lib/dependabot/python/python_versions.rb [hunk 6]
```diff
@@ -4,14 +4,14 @@ module Dependabot
...
       ).freeze
 
       # Due to an OpenSSL issue we can only install the following versions in
       # the Dependabot container.
       SUPPORTED_VERSIONS = %w(
-        3.9.2 3.9.1 3.9.0
-        3.8.8 3.8.7 3.8.6 3.8.5 3.8.4 3.8.3 3.8.2 3.8.1 3.8.0
+        3.9.3 3.9.2 3.9.1 3.9.0
```
Comment (Parnassius) [inline python/lib/dependabot/python/python_versions.rb:13]:
```suggestion
        3.9.4 3.9.2 3.9.1 3.9.0
```

File: python/spec/dependabot/python/update_checker/pipenv_version_resolver_spec.rb [hunk 7]
```diff
@@ -255,7 +255,7 @@
                 to start_with("Dependabot detected the following Python")
               expect(error.message).to include("3.4.*")
               expect(error.message).
-                to include("supported in Dependabot: 3.9.2, 3.9.1, 3.9.0")
+                to include("supported in Dependabot: 3.9.3, 3.9.2, 3.9.1")
```
Comment (Parnassius) [inline python/spec/dependabot/python/update_checker/pipenv_version_resolver_spec.rb:258]:
```suggestion
                to include("supported in Dependabot: 3.9.4, 3.9.2, 3.9.1")
```

**Review by Parnassius — CHANGES_REQUESTED** [review]
In the meantime python 3.9.3 was recalled due to an ABI incompatibility, and was replaced by [3.9.4](https://www.python.org/downloads/release/python-394/). Pyenv also [made a new release](https://github.com/pyenv/pyenv/releases/tag/1.2.26) to support it

**Review by jurre — APPROVED** [review]
Thanks for adding this! I noticed from the PR history that adding a python version is quite painful if you're not intimately familiar wit our setup. Would you mind sticking a small note in the README explaining how to do it for the next person?

**Comment by sobolevn** [comment]
Sure!

You mean this `README`? https://github.com/dependabot/dependabot-core/blob/main/python/README.md
Or the main one?

**Comment by jurre** [comment]
> Sure!
> 
> You mean this `README`? https://github.com/dependabot/dependabot-core/blob/main/python/README.md
> Or the main one?

Wherever you would have hoped to find it 😄 I think the main one might be easier for visibility?

**Comment by sobolevn** [comment]
It looks like `python/README.md` is more intuitive.

**Comment by jurre** [comment]
Thanks @sobolevn 💛

**Pull request merged**